src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -124,12 +156,9 @@ internal static CounterGroup GetCounterGroup(EventSource eventSource)
 
         private void EnableTimer(float pollingIntervalInSeconds)
         {
+            Debug.Assert(pollingIntervalInSeconds > 0);
             Debug.Assert(Monitor.IsEntered(s_counterGroupLock));
-            if (pollingIntervalInSeconds <= 0)
```

Comment:
Given the assert, this part of the change looks like non-essential dead-code elimination. I'd suggest not making this part of the change.